include/net/socket.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __NET_SOCKET_H
+#define __NET_SOCKET_H
+
+#include <sys/types.h>
+#include <zephyr/types.h>
+#include <net/net_ip.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int zsock_socket(int family, int type, int proto);
+int zsock_close(int sock);
+int zsock_bind(int sock, const struct sockaddr *addr, socklen_t addrlen);
+int zsock_connect(int sock, const struct sockaddr *addr, socklen_t addrlen);
+int zsock_listen(int sock, int backlog);
+int zsock_accept(int sock, struct sockaddr *addr, socklen_t *addrlen);
+ssize_t zsock_send(int sock, const void *buf, size_t len, int flags);
+ssize_t zsock_recv(int sock, void *buf, size_t max_len, int flags);
```

Comment:
A few reasons:
  1. It's often a standard module convention to have symbols declared in their own header.  Makes it easy to find things.  (I know, Zephyr doesn't always follow this convention).
  2. These are Zephyr APIs, which may have divergence from true POSIX APIs (declared in socket.h).
  3.  It makes it easier to implement a new socket provider for the socket APIs for offloading.  zsocket could just be one (the default) socket provider.  Socket.h  (or socket.c) would be the switch layer.

I assumed zsocket.h would also go into net/.